### PR TITLE
feat(config): default app permissions (iOS)

### LINKED ISSUE #2978: [Enhancement] Support for default permissions
**Is your feature request related to a problem? Please describe.**

I just added a new package (https://github.com/zo0r/react-native-push-notification) an it became tedious to specify everywhere `launchApp` is specified `{ permissions: { notifications: 'NO'} }` so the pop-up won't show when the app is starting.

**Describe the solution you'd like**
Add support for default permissions that could be overridden when specifying `permissions`.
This could be either in `detox.init` or in the configuration file.

**Describe alternatives you've considered**
Specifying `{ permissions: { notifications: 'NO'} }` everywhere `launchApp` is specified.

### PR DESCRIPTION
**Pull request by noomorph**
## Description

Resolves #2978.

> _For features/enhancements:_
 - [x] I have added/updated the relevant references in the [documentation](https://github.com/wix/Detox/tree/master/docs) files.

> _For API changes:_
 - [x] I have made the necessary changes in the [types index](https://github.com/wix/Detox/blob/master/detox/index.d.ts) file.

### CODE REVIEW

**Comment by noomorph** [comment]
I see Docusaurus build error – I'll fix tomorrow.

File: detox/src/devices/runtime/RuntimeDevice.js [hunk 1]
```diff
@@ -189,9 +192,13 @@ class RuntimeDevice {
...
   async installApp(binaryPath, testBinaryPath) {
-    await traceCall('appInstall', () => {
+    await traceCall('appInstall', async () => {
+      /** @type {*} */
       const currentApp = binaryPath ? { binaryPath, testBinaryPath } : this._getCurrentApp();
-      return this.deviceDriver.installApp(currentApp.binaryPath, currentApp.testBinaryPath);
+      await this.deviceDriver.installApp(currentApp.binaryPath, currentApp.testBinaryPath);
+      if (!_.isEmpty(currentApp.permissions)) {
+        await this.deviceDriver.setPermissions(currentApp.bundleId, currentApp.permissions);
+      }
```
Comment (d4vidi) [inline detox/src/devices/runtime/RuntimeDevice.js:201]: Well I didn't go deeply into this but something about this approach feels a bit off. It assumes that the permissions can be set in such a global way - before an instance of the app is running. While it works on iOS, I can't 100% say the same about Android / a theoretical future platform.

What about applying a similar solution to the launch-args technique, which is set in advance and handled on-site (i.e. in launch app)?

File: detox/index.d.ts [hunk 2]
```diff
@@ -201,6 +202,7 @@ declare global {
             build?: string;
             testBinaryPath?: string;
             launchArgs?: Record<string, any>;
+            permissions?: DevicePermissions;
```
Comment (d4vidi) [inline detox/index.d.ts:205]: Redundant?

File: docs/APIRef.DeviceObjectAPI.md [hunk 3]
```diff
@@ -528,3 +530,32 @@ Exposes [`UiAutomator`’s `UiDevice` API](https://developer.android.com/referen
...
+Detox config file or [launching the app](#2-permissionsset-runtime-permissions-ios-only),
+you can give certain permissions beforehand to it to avoid this inconvenient modal:
+
+![iOS permissions modal](img/ios-permissions.png)
+
+So, here is the exhaustive permissions list that Detox currently supports for iOS:
+
+| Permission    | Values                                                  |
+|---------------|---------------------------------------------------------|
+| calendar      | `'YES' &#124; 'NO' &#124; 'unset'`                      |
```
Comment (d4vidi) [inline docs/APIRef.DeviceObjectAPI.md:546]: `&124;` doesn't seem to render well

File: docs/APIRef.DeviceObjectAPI.md [hunk 4]
```diff
@@ -528,3 +530,32 @@ Exposes [`UiAutomator`’s `UiDevice` API](https://developer.android.com/referen
 **This is not a part of the official Detox API**, it may break and change whenever an update to `UiDevice` or `UiAutomator` Gradle dependencies (`androidx.test.uiautomator:uiautomator`) is introduced.
 
 [`UiDevice`’s autogenerated code](https://github.com/wix/Detox/tree/a9a09246c05733f6b91cfcc0dba05a4714abca92/detox/src/android/espressoapi/UIDevice.js)
+
+### Permissions (iOS only)
```
Comment (d4vidi) [inline docs/APIRef.DeviceObjectAPI.md:534]: Add link to ToC

File: docs/APIRef.DeviceObjectAPI.md [hunk 5]
```diff
@@ -528,3 +530,32 @@ Exposes [`UiAutomator`’s `UiDevice` API](https://developer.android.com/referen
...
+
+When you are [configuring the apps](APIRef.Configuration.md#apps-configurations) list in your
+Detox config file or [launching the app](#2-permissionsset-runtime-permissions-ios-only),
+you can give certain permissions beforehand to it to avoid this inconvenient modal:
```
Comment (d4vidi) [inline docs/APIRef.DeviceObjectAPI.md:538]: A simple and elegant explanation

**Pull request closed without merging**